marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/RowBatcherImpl.java

```diff
@@ -0,0 +1,312 @@
+        return this;
+    }
+
+    @Override
+    public RowBatcher withJobId(String jobId) {
+        this.jobId = jobId;
+        return this;
+    }
+
+    @Override
+    public String getJobId() {
+        return this.jobId;
+    }
+
+    @Override
+    public RowBatcher withJobName(String jobName) {
+        this.jobName = jobName;
+        return this;
+    }
+
+    @Override
+    public String getJobName() {
+        return this.jobName;
+    }
+
+    @Override
+    public RowBatcher withConsistentSnapshot() {
+
+        return this;
+    }
+
+    @Override
+    public RowBatchSuccessListener[] getSuccessListeners() {
+        return new RowBatchSuccessListener[0];
+    }
+
+    @Override
+    public RowBatchFailureListener[] getFailureListeners() {
+        return failureListeners.toArray(new RowBatchFailureListener[failureListeners.size()]);
+    }
+
+    @Override
+    public void setSuccessListeners(RowBatchSuccessListener[] listeners) {
+        this.listeners = listeners;
+    }
+
+    @Override
+    public void setFailureListeners(RowBatchFailureListener... listeners) {
+
+    }
+
+    @Override
+    public boolean awaitCompletion() {
+        try {
+            return awaitCompletion(Long.MAX_VALUE, TimeUnit.DAYS);
+        } catch(InterruptedException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean awaitCompletion(long timeout, TimeUnit unit) throws InterruptedException {
+        if ( threadPool == null ) {
+            throw new IllegalStateException("Job not started.");
+        }
+        return threadPool.awaitTermination(timeout, unit);
+    }
+
+    @Override
+    public void retry(RowBatchRequestEvent event) {
+        if ( isStopped() == true ) {
+            logger.warn("Job is now stopped, aborting the retry");
+            return;
+        }
+        start(getJobTicket());
+    }
+
+    @Override
+    public void retryWithFailureListeners(RowBatchRequestEvent event) {
+        retry(event);
+    }
+
+    @Override
+    public JobTicket getJobTicket() {
+        return this.jobTicket;
+    }
+
+    @Override
+    public void stop() {
+        stopped.set(true);
+        if ( threadPool != null ) threadPool.shutdownNow();
+        jobEndTime = Calendar.getInstance();
+
+    }
+
+    @Override
+    public Calendar getJobStartTime() {
+        return this.jobStartTime;
+    }
+
+    @Override
+    public Calendar getJobEndTime() {
+        return this.jobEndTime;
+    }
+
+    @Override
+    public DatabaseClient getPrimaryClient() {
+        return null;
+    }
+
+    @Override
+    public void start(JobTicket ticket) {
+        if ( threadPool != null ) {
+            logger.warn("startJob called more than once");
+            return;
+        }
+        if ( getBatchSize() <= 0 ) {
+            withBatchSize(1);
+            logger.warn("batchSize should be 1 or greater--setting batchSize to 1");
+        }
+        jobTicket = ticket;
+
+        getRowCount();
```

Comment:
Once we know the batch size, we can spin up the threads and submit tasks.

As sketched in the technical story, each task should 

1. get the next lower bound
2. calculate the upper bound
3. bind the parameters on the plan to the lower and upper bound values
4. use the RowManager to get the result rows
5. send the result rows to the success listener

This code will be similar to startQuerying() and QueryTask in QueryBatcherImpl